modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/translator/DeepLTranslator.java

```diff
@@ -0,0 +1,243 @@
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.module.configuration.ConfigurationException;
+import com.liferay.portal.kernel.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.translation.exception.TranslatorException;
+import com.liferay.translation.translator.Translator;
+import com.liferay.translation.translator.TranslatorPacket;
+import com.liferay.translation.translator.deepl.internal.configuration.DeepLTranslatorConfiguration;
+import com.liferay.translation.translator.deepl.internal.constants.DeepLConstants;
+import com.liferay.translation.translator.deepl.internal.model.SupportedLanguage;
+import com.liferay.translation.translator.deepl.internal.model.TranslateResponse;
+import com.liferay.translation.translator.deepl.internal.model.Translation;
+
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Yasuyuki Takeo
+ */
+@Component(
+	configurationPid = "com.liferay.translation.translator.deepl.internal.configuration.DeepLTranslatorConfiguration",
+	service = Translator.class
+)
+public class DeepLTranslator implements Translator {
+
+	@Override
+	public boolean isEnabled(long companyId) throws ConfigurationException {
+		DeepLTranslatorConfiguration deepLTranslatorConfiguration =
+			_configurationProvider.getCompanyConfiguration(
+				DeepLTranslatorConfiguration.class, companyId);
+
+		if (deepLTranslatorConfiguration.enabled() &&
+			!Validator.isBlank(deepLTranslatorConfiguration.authKey()) &&
+			!Validator.isBlank(deepLTranslatorConfiguration.url())) {
+
+			return true;
+		}
+
+		return false;
+	}
+
+	@Override
+	public TranslatorPacket translate(TranslatorPacket translatorPacket)
+		throws PortalException {
+
+		if (!isEnabled(translatorPacket.getCompanyId())) {
+			return translatorPacket;
+		}
+
+		DeepLTranslatorConfiguration deepLTranslatorConfiguration =
+			_configurationProvider.getCompanyConfiguration(
+				DeepLTranslatorConfiguration.class,
+				translatorPacket.getCompanyId());
+
+		String tlc = translatorPacket.getTargetLanguageId();
+
+		String targetLanguageCode = _getLanguageCode(tlc);
+
+		List<String> supportedLanguages = _getSupportedLanguages(
+			deepLTranslatorConfiguration);
+
+		if (!_verifyLanguage(supportedLanguages, targetLanguageCode)) {
+			String sls = StringUtil.merge(
```

Comment:
Fully spell variable names. Also, please review your code, look for this kind of variables and rename them (I won't mention further occurrences of this issue).